shell/shared/db.js

```diff
@@ -86,6 +86,46 @@ Grains = new Mongo.Collection("grains");
 //   publicId:  An id used to publicly identify this grain. Used e.g. to route incoming e-mail and
 //       web publishing. This field is initialized when first requested by the app.
 
+RoleAssignments = new Mongo.Collection("roleAssignments");
+// Edges in the permissions sharing graph.
+//
+// To share a permission with another user is to declare "if I have this permission, then so should
+// the recipient". A bundle of such declarations by a single sharer directed at a single recipient
+// for a single grain is called a "role assignment". A grain's owner always has every permission,
+// but permissions for other users must be computed from this collection.
+//
+// Every role assignment implicitly shares the special "can view grain" permission, which dictates
```

Comment:
I don't think it's a big deal to leak the fact that a grain was deleted to users who at one point had access but were later revoked, so let's not worry about that.

Separately from that, though, this comment as it stands still seems to be saying that "can open a session" and "shows up in grain list" are the same thing. But in the Alice -> Bob -> Carol case, after Bob's access is revoked, Carol cannot open a session, but will still see the capability in her grain list unless we transitively delete edges, which we don't want to do.